sdks/java/extensions/euphoria/euphoria-beam/src/main/java/org/apache/beam/sdk/extensions/euphoria/beam/BeamAccumulatorProvider.java

```diff
@@ -0,0 +1,160 @@
+import java.util.stream.LongStream;
+import org.apache.beam.sdk.extensions.euphoria.core.client.accumulators.AccumulatorProvider;
+import org.apache.beam.sdk.extensions.euphoria.core.client.accumulators.Counter;
+import org.apache.beam.sdk.extensions.euphoria.core.client.accumulators.Histogram;
+import org.apache.beam.sdk.extensions.euphoria.core.client.accumulators.Timer;
+import org.apache.beam.sdk.extensions.euphoria.core.util.Settings;
+import org.apache.beam.sdk.metrics.Metrics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Translation of accumulators to {@link Metrics}. Metrics from same type of accumulator will have
+ * same namespace and will differ only in given name. */
+public class BeamAccumulatorProvider implements AccumulatorProvider {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BeamAccumulatorProvider.class);
+  private final Map<String, Counter> counterMap;
+  private final Map<String, Timer> timerMap;
+  private final Map<String, Histogram> histogramMap;
+
+  private BeamAccumulatorProvider() {
+    counterMap = new ConcurrentHashMap<>();
+    timerMap = new ConcurrentHashMap<>();
+    histogramMap = new ConcurrentHashMap<>();
+  }
+
+  public static Factory getFactory() {
+    return Factory.get();
+  }
+
+  @Override
+  public Counter getCounter(String name) {
+    return counterMap.computeIfAbsent(name, BeamMetricsCounter::new);
+  }
+
+  @Override
+  public Histogram getHistogram(String name) {
+    return histogramMap.computeIfAbsent(name, BeamMetricsHistogram::new);
+  }
+
+  @Override
+  public Timer getTimer(String name) {
+    return timerMap.computeIfAbsent(name, BeamMetricsTimer::new);
+  }
+
+  // ------------------------
+
+  /** AccumulatorProvider Factory. */
+  public static class Factory implements AccumulatorProvider.Factory {
+
+    private static final BeamAccumulatorProvider.Factory INSTANCE =
+        new BeamAccumulatorProvider.Factory();
+
+    private static final AccumulatorProvider PROVIDER = new BeamAccumulatorProvider();
+
+    private static final AtomicBoolean isLogged = new AtomicBoolean();
+
+    private Factory() {}
+
+    public static BeamAccumulatorProvider.Factory get() {
+      return INSTANCE;
+    }
+
+    @Override
+    public AccumulatorProvider create(Settings settings) {
+      if (isLogged.compareAndSet(false, true)) {
+        LOG.warn("Using accumulators with BeamAccumulatorProvider");
+      }
+      return PROVIDER;
+    }
+  }
+
+  // ------------------------
+
+  /** Implementation of Counter via {@link org.apache.beam.sdk.metrics.Counter}. */
+  public static class BeamMetricsCounter extends BeamMetrics implements Counter {
+
+    public BeamMetricsCounter(String name) {
+      super(name);
+    }
+
+    @Override
+    public void increment(long value) {
+      Metrics.counter(this.getClass(), getName()).inc(value);
+    }
+
+    @Override
+    public void increment() {
+      Metrics.counter(this.getClass(), getName()).inc();
```

Comment:
Several ideas come to my mind:
* It would be nice to have Flows/Pipelines/Jobs named and use their name as namespace.
* Job run id. If possible to acquire it.